extensions/topic_extension.rb

```diff
@@ -95,7 +95,8 @@ def qa_votes(topic, user)
 
       def qa_enabled(topic)
         return false unless SiteSetting.qa_enabled
-        return false if !topic || topic.id == topic.category.topic_id
+        return false if !topic
+        return false if topic.category && topic.category.topic_id == topic.id
```

Comment:
```suggestion
        return false if topic.category&.topic_id == topic.id
```

Reply:
I intentionally chose not to do this so that we don't execute the comparison.